pilopt/src/lib.rs

```diff
@@ -620,3 +637,66 @@ fn remove_duplicate_identities<T: FieldElement>(pil_file: &mut Analyzed<T>) {
         .collect();
     pil_file.remove_identities(&to_remove);
 }
+
+fn equal_constrained<T: FieldElement>(
+    poly_data: (&u64, &AlgebraicExpression<T>),
```

Comment:
Why not two separate parameters (I think it would also make it clearer what `poly_data` is).